src/realm/object-store/impl/collection_notifier.hpp

```diff
@@ -36,6 +36,35 @@
 namespace realm {
 namespace _impl {
 
+// A `NotificationCallback` is added to a collection when observing it.
+// It contains all information necessary in case we need to notify about changes
+// to this collection.
+struct NotificationCallback {
+    // The callback function being invoked when we notify for changes in this collection.
+    CollectionChangeCallback fn = CollectionChangeCallback();
```

Comment:
Initializing member variables like this is redundant and very unidiomatic in c++.

Reply:
This change was made based on a suggestion by @jbreams. I probably didn't solve it how he expected it.
Can the two of you clarify what you'd prefer here?

Reply:
What I meant was to initialize POD members (I know the term POD is deprecated in C++20, don't @ me) where you'll use the default constructor of the class they are a member of. Classes and struct members don't need to be initialized like this - they get initialized automatically via their default constructor.

I come from a C++ background where you're encouraged to be explicit tho. Don't assume `bool b;` is false, set it to false explicitly so you _know_ it's false.

Reply:
I agree with @tgoyne. Explicit initialization is redundant where you have a default constructor.

Reply:
Going with the majority here so that we can make progress with this PR.
Reverted with ceb447df5.